src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -13,25 +13,509 @@ module Algebra.Graph.Relation.Symmetric (
     -- * Data structure
     SymmetricRelation, fromRelation, toRelation,
 
+    -- * Basic graph construction primitives
+    empty, vertex, edge, overlay, connect, vertices, edges, overlays, connects,
+
+    -- * Relations on graphs
+    isSubgraphOf,
+
     -- * Graph properties
-    neighbours
+    isEmpty, hasVertex, hasEdge, vertexCount, edgeCount, vertexList, edgeList,
+    adjacencyList, vertexSet, edgeSet, preSet, postSet, neighbours,
+
+    -- * Standard families of graphs
+    path, circuit, clique, biclique, star, stars, tree, forest,
+
+    -- * Graph transformation
+    removeVertex, removeEdge, replaceVertex, mergeVertices, transpose, gmap, induce,
+
+    -- * Relational operations
+    compose, -- closure, reflexiveClosure, symmetricClosure, transitiveClosure
   ) where
 
-import Algebra.Graph.Relation
+import qualified Algebra.Graph.Relation as R
+import qualified Algebra.Graph.Relation.Internal as RI
 import Algebra.Graph.Relation.InternalDerived
+import Algebra.Graph.Class (connect, empty, vertex, overlay)
 
 import qualified Data.Set as Set
+import Data.Tuple
+import Data.Tree 
+import qualified Data.Tree as Tree
 
 -- | Construct a symmetric relation from a 'Relation'.
 -- Complexity: /O(1)/ time.
-fromRelation :: Relation a -> SymmetricRelation a
+fromRelation :: R.Relation a -> SymmetricRelation a
 fromRelation = SymmetricRelation
 
 -- | Extract the underlying relation.
 -- Complexity: /O(m*log(m))/ time.
-toRelation :: Ord a => SymmetricRelation a -> Relation a
-toRelation = symmetricClosure . fromSymmetric
+toRelation :: Ord a => SymmetricRelation a -> R.Relation a
+toRelation = R.symmetricClosure . fromSymmetric
+
+-- | Construct the graph comprising /a single edge/.
+-- Complexity: /O(1)/ time, memory and size.
+--
+-- @
+-- edge x y               == 'overlay' ('connect' ('vertex' x) ('vertex'
+-- y)) ('connect' ('vertex' y) ('vertex' x))
+-- 'hasEdge' x y (edge x y) == True
+-- 'edgeCount'   (edge x y) == 2
+-- 'vertexCount' (edge 1 1) == 1
+-- 'vertexCount' (edge 1 2) == 2
+-- @
+edge :: Ord a => a -> a -> SymmetricRelation a
+edge x y = SymmetricRelation $ RI.Relation (Set.fromList [x, y]) (Set.fromList [(x, y), (y,x)]) 
```

Comment:
You should also provide primitives `empty`, `vertex`, `overlay`, and `connect`.

Reply:
I have those primitives in the `InternalDerived`module when implementing the `Graph` should I do it differently?

Reply:
We typically export monomorphised versions of all primitive constructions for documentation purposes. See an example here: https://github.com/snowleopard/alga/blob/6777032aaab79b7c6128f04ad2c1bf8a48659496/src/Algebra/Graph/AdjacencyMap.hs#L56-L66

Reply:
Got it!

Reply:
I still don't see these primitives.

Reply:
Sorry, never mind, you do export them!